web-api/src/business/useCases/trialSessions/generateSuggestedTrialSessionCalendarInteractor.ts

```diff
@@ -0,0 +1,336 @@
+  sessions,
+  termEndDate,
+  termStartDate,
+}: {
+  sessions: RawTrialSession[];
+  termEndDate: string;
+  termStartDate: string;
+}): RawTrialSession[] => {
+  return sessions.filter(session => {
+    const isSessionInTerm = isDateWithinGivenInterval({
+      date: session.startDate,
+      intervalEndDate: formatDateString(termEndDate, FORMATS.ISO),
+      intervalStartDate: formatDateString(termStartDate, FORMATS.ISO),
+    });
+
+    return (
+      session.sessionType === SESSION_TYPES.special &&
+      session.isCalendared &&
+      session.sessionStatus !== SESSION_STATUS_TYPES.closed &&
+      isSessionInTerm
+    );
+  });
+};
+
+export const getPreviousTwoTerms = (termStartDate: string) => {
+  const { month, year } = deconstructDate(termStartDate);
+
+  const currentTerm = getCurrentTermByMonth(month);
+  const terms = [
+    `spring, ${+year - 1}`,
+    `fall, ${+year - 1}`,
+    `winter, ${year}`,
+    `spring, ${year}`,
+    `fall, ${year}`,
+  ];
+  const termIndex = terms.findIndex(t => `${currentTerm}, ${year}` === t);
+  const term1 = terms[termIndex - 1];
+  const term2 = terms[termIndex - 2];
+
+  return [term1, term2];
+};
+
+const SESSION_TERMS_FOR_GENERATOR = {
+  fall: [9, 10, 11, 12],
+  spring: [4, 5, 6, 7, 8],
+  winter: [1, 2, 3],
+};
+
+export const getCitiesFromLastTwoTerms = ({
+  sessions,
+  termStartDate,
+}: {
+  sessions: RawTrialSession[];
+  termStartDate: string;
+}): string[] => {
+  const previousTwoTerms = getPreviousTwoTerms(termStartDate);
+  return sessions
+    .filter(session => {
+      const termString = `${session.term.toLowerCase()}, ${session.termYear}`;
+      return (
+        previousTwoTerms.includes(termString) &&
+        session.sessionType !== SESSION_TYPES.special
+      );
+    })
+    .map(relevantSession => {
+      return relevantSession.trialLocation!;
+    });
+};
+
+const generateMessage = ({
+  userMessages,
+}: {
+  userMessages: string[];
+}): CalendarGeneratorMessage => {
+  if (userMessages.length === 0) {
+    return {
+      message: SUGGESTED_TRIAL_SESSION_TITLES.success,
+      title: undefined,
+      type: MESSAGE_TYPES.success,
+    };
+  } else {
+    let messageString = '';
+    userMessages.forEach(userMessage => {
+      messageString = messageString + ' ' + userMessage;
+    });
+    messageString = messageString.trim();
+
+    return {
+      message: messageString,
+      title: SUGGESTED_TRIAL_SESSION_TITLES.warning,
+      type: MESSAGE_TYPES.warning,
+    };
+  }
+};
+
+const calendarIsEmpty = (
+  caseCountsAndSessionsByCity: CaseCountsAndSessionsByCity,
+) => {
+  return Object.values(caseCountsAndSessionsByCity).every(cityObject => {
+    return cityObject.scheduledSessions.length < 1;
+  });
+};
+
+const getCurrentTermByMonth = (currentMonth: string): string => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const term = Object.entries(SESSION_TERMS_FOR_GENERATOR).find(([_, months]) =>
```

Comment:
I think you can do
`const term = Object.entries(SESSION_TERMS_FOR_GENERATOR).find(([_, months])`
and avoid the eslint disable line

Reply:
Doh! I meant `const term = Object.entries(SESSION_TERMS_FOR_GENERATOR).find(([, months])`--without the underscore. Sorry for the confusion!